In [6]:
import tensorflow as tf
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten
from tensorflow.keras.utils import to_categorical

# 1. Load dataset MNIST
(X_train, y_train), (X_test, y_test) = mnist.load_data()

# Normalisasi data (0-255 → 0-1)
X_train = X_train / 255.0
X_test = X_test / 255.0

# One-hot encoding label
y_train = to_categorical(y_train, 10)
y_test = to_categorical(y_test, 10)

# 2. Bangun model JST
model = Sequential([
    Flatten(input_shape=(28, 28)),      # Ubah gambar 28x28 menjadi vektor
    Dense(256, activation='relu'),      # Hidden layer 1
    Dense(128, activation='relu'),       # Hidden layer 2
    Dense(128, activation='relu'),       # Hidden layer 3
    Dense(10, activation='softmax')     # Output layer (10 kelas)
])

# 3. Kompilasi model
model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

# 4. Latih model

# 5. Evaluasi model
loss, acc = model.evaluate(X_test, y_test)
print(f"Akurasi pada data uji: {acc:.4f}")

313/313 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.0878 - loss: 2.3355  
Akurasi pada data uji: 0.0878


| Model                        | Loss     | Akurasi   |
|------------------------------------------|----------|-----------|
| Model Awal (128, 64 neuron, ReLU)        | 2.3401   | 0.0980    |
| Model Hidden Layer 3 Neuron (ReLU)       | 2.3355   | 0.0878    |